main.py

```diff
@@ -0,0 +1,49 @@
+import requests
+from dotenv import dotenv_values
+
+
+def get_metadados(token):
+    my_headers = {'Authorization' : f'Bearer {token}'}
+    response = requests.get('https://api.github.com/user/repos', headers=my_headers)
+    resp = response.json()
+    metadados = [{'name':repo['name'],
+                  'full_name':repo['full_name'],
+                  'is_private':repo['private']} for repo in resp]
+    with open('repos/metadados.txt', 'w') as f:
+        for line in metadados:
+            f.write(str(line)+'\n')
+    return metadados 
+
+def download_repos(token, EXT='zip'):
+    #EXT  = 'tar'  # it also works
+    
+    metadados = get_metadados(token)
+    headers = {
+        "Authorization" : f'token {token}',
+        "Accept": 'application/vnd.github.v3+json'
+    }
+    
+    REF  = ''      # master/main branch 
+    
+    
+    
+    full_names = [repo['full_name'] for repo in metadados]
+    for owner_repo  in full_names:
+        repo_name = owner_repo.split('/')[-1]
+        url = f'https://api.github.com/repos/{owner_repo}/{EXT}ball/{REF}'
+        r = requests.get(url, headers=headers)
+        if r.status_code == 200:
```

Comment:
This logic is OK but you should do that differently. Create a try/except block instead of if/else and raise an error if something bad occurs, take a look at [this](https://www.adamsmith.haus/python/answers/how-to-catch-an-httperror-in-python) and [this](https://www.geeksforgeeks.org/response-raise_for_status-python-requests/).